### ISSUE TITLE
Random seed works at 0.15.1 segment_plane  but not works at 0.16.0

### ISSUE THREAD

**Issue by mcmingchang** [isl-org/Open3D#5647]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

Random seed works at 0.15.1 segment_plane  but not works at 0.16.0

### Steps to reproduce the bug

```python
#### in 0.15.1
import numpy as np
import open3d as o3d
pcd = o3d.io.read_point_cloud('1.ply')
pcd_down = pcd.voxel_down_sample(voxel_size=0.01)
plane_model, inliers = pcd_down.segment_plane(distance_threshold=0.02,
                                              ransac_n=5,
                                              num_iterations=5000, seed=1)
plane_model, inliers2 = pcd_down.segment_plane(distance_threshold=0.02,
                                              ransac_n=5,
                                              num_iterations=5000, seed=1)
print(np.all(np.array(inliers) == np.array(inliers2)))


##### in 0.16.0
import numpy as np
import open3d as o3d
o3d.utility.random.seed(1)
pcd = o3d.io.read_point_cloud('1.ply')
pcd_down = pcd.voxel_down_sample(voxel_size=0.01)
plane_model, inliers = pcd_down.segment_plane(distance_threshold=0.02,
                                              ransac_n=5,
                                              num_iterations=5000)
plane_model, inliers2 = pcd_down.segment_plane(distance_threshold=0.02,
                                              ransac_n=5,
                                              num_iterations=5000)
print(np.all(np.array(inliers) == np.array(inliers2)))
```

### Error message

None

### Expected behavior

The result of 0.16 should be the same as that of 0.15

### Open3D, Python and System information

```markdown
None
```

### Additional information

Random seed works at 0.15.1 segment_plane  but not works at 0.16.0

**Comment by yuecideng** [comment]
Since 0.16.0, the `segment_plane` function is parallel and the iteration is changed with the estimated inliers ratio, which may lead to inconsistent results under the same random seed.

**Comment by mcmingchang** [comment]
You mean it's not a bug？But setting seed is to achieve a stable result.

**Comment by yuecideng** [comment]
You may try to set the new arg `probability=1`, which will force the iteration to be executed equals to `num_iterations`.  Then the results will be consistent with the seed you set.

**Comment by mcmingchang** [comment]
After experiments, the results are different each time.

**Comment by yuecideng** [comment]
The random seed is singleton so you should reset the random seed before `segment_plane` at the second time. 
```python
import numpy as np
import open3d as o3d
o3d.utility.random.seed(1)
pcd = o3d.io.read_point_cloud('1.ply')
pcd_down = pcd.voxel_down_sample(voxel_size=0.01)
plane_model, inliers = pcd_down.segment_plane(distance_threshold=0.02,
                                              ransac_n=5,
                                              num_iterations=5000,
                                              probability=1)
o3d.utility.random.seed(1)
plane_model, inliers2 = pcd_down.segment_plane(distance_threshold=0.02,
                                              ransac_n=5,
                                              num_iterations=5000,
                                              probability=1)
print(np.all(np.array(inliers) == np.array(inliers2)))
```

**Comment by mcmingchang** [comment]
thank you

**Comment by mcmingchang** [comment]
For large plane fitting, it is not work at 0.16

**Comment by conorsim** [comment]
I have also observed this issue with 0.16 and 0.17

**Comment by TheExDeus** [comment]
I set seed before each call and probability=1.0 - and it still doesn't make this deterministic. I also get different value every time (and in some cases it can vary a lot if the points are sparse). This means we cannot make tests. Any idea on how to fix this?

This means I might need to revert to 0.15, but that doesn't seem like a long term solution.

**Comment by yuecideng** [comment]
Hi @TheExDeus @conorsim, could you provide the data or the code snippet to reproduce the bug?

**Comment by conorsim** [comment]
@yuecideng if I test your code above with version 0.17.0 I don't get the same values. I made a couple of changes:

1. To initialize a point cloud of random points since I don't have a .ply file handy
2. To compare `plane_model` since `inliers` doesn't always have the same shape (due to the this issue)

```
import numpy as np
import open3d as o3d

pcd = o3d.geometry.PointCloud()
pcd.points = o3d.utility.Vector3dVector(np.random.uniform(1,10,(1000,3)))

for _ in range(10):
    o3d.utility.random.seed(1)
    pcd_down = pcd.voxel_down_sample(voxel_size=0.01)
    plane_model, inliers = pcd_down.segment_plane(distance_threshold=0.02,
                                                  ransac_n=5,
                                                  num_iterations=5000,
                                                  probability=1)
    o3d.utility.random.seed(1)
    plane_model2, inliers2 = pcd_down.segment_plane(distance_threshold=0.02,
                                                  ransac_n=5,
                                                  num_iterations=5000,
                                                  probability=1)
    print(np.all(np.array(plane_model) == np.array(plane_model2)))
```

When I run this, the output I got was
```
False
False
False
False
True
False
False
False
False
False
```
So, in this run it was correct only 1/10 times.

**Comment by yuecideng** [comment]
Hi @conorsim, you are right! this may be caused by the update of best estimated model among multi threading. I will try to investigate it.

**Comment by yuecideng** [comment]
The issue has been solved by PR #6308.
